projects/step-core/src/lib/modules/tree/components/tree/tree.component.ts

```diff
@@ -73,14 +75,15 @@ export class TreeComponent<N extends TreeNode> implements TreeNodeTemplateContai
     event.preventDefault();
     this.contextMenuPosition.x = event.clientX;
     this.contextMenuPosition.y = event.clientY;
-    this.contextMenuTrigger.menuData = { node };
+    this.contextMenuTrigger.menuData = { node, multipleNodes };
     this.contextMenuTrigger.openMenu();
     this.openedMenuNodeId = nodeId;
   }
 
   handleContextAction(action: TreeAction, node?: N): void {
     const actionId = action.id;
-    this.treeContextAction.emit({ actionId, node });
+    const multipleNodes = this.contextMenuTrigger.menuData.multipleNodes;
+    this.treeContextAction.emit({ actionId, node, multipleNodes });
```

Comment:
That's fine. As alternative multipleNodes could be passed as parameter of `handleContectAction`